code/modules/clothing/glasses/glasses.dm

```diff
@@ -46,6 +45,16 @@
 		"Grey" = 'icons/mob/species/grey/eyes.dmi'
 		)
 
+/obj/item/clothing/glasses/meson/attack_self()
+	visor_flags = !visor_flags
+	if (visor_flags)
+		invis_view = SEE_INVISIBLE_MINIMUM //don't render darkness while wearing these
+		vision_flags = SEE_TURFS

```

Comment:
You need to ensure sight is updated, otherwise, this is going to mess up.

I could see hypothetical problems arising from this by cleverly toggling things on/off at the right time with the right vision flags to abuse this, otherwise.